compliance/compliance.go

```diff
@@ -537,6 +555,135 @@ func ModifyConnectionSinglePrimaryPreserve(c *fluent.GRIBIClient, t testing.TB,
 	chk.HasRecvClientErrorWithStatus(t, err, want, chk.AllowUnimplemented())
 }
 
+// InvalidElectionIDAndAFTOperation ensures that the server returns an error when the client
+// attempts to update the election ID whilst simultaenously specifying an operation.
+func InvalidElectionIDAndAFTOperation(c *fluent.GRIBIClient, t testing.TB, _ ...TestOpt) {
+	defer electionID.Inc()
+	c.Connection().WithRedundancyMode(fluent.ElectedPrimaryClient).WithPersistence().WithInitialElectionID(electionID.Load(), 0)
+	c.Start(context.Background(), t)
+	defer c.Stop(t)
+
+	c.StartSending(context.Background(), t)
+
+	// Inject a specfic invalid entry that specifies election ID in two places along
+	// with an entry.
+	c.Modify().InjectRequest(t, &spb.ModifyRequest{
+		ElectionId: &spb.Uint128{
+			Low:  electionID.Load(),
+			High: 0,
+		},
+		Operation: []*spb.AFTOperation{{
+			ElectionId: &spb.Uint128{
+				Low:  electionID.Load(),
+				High: 0,
+			},
+			Id:              1,
+			NetworkInstance: defaultNetworkInstanceName,
+			Op:              spb.AFTOperation_ADD,
+			Entry: &spb.AFTOperation_NextHop{
+				NextHop: &aftpb.Afts_NextHopKey{
+					Index:   1,
+					NextHop: &aftpb.Afts_NextHop{},
+				},
+			},
+		}},
+	})
+
+	err := awaitTimeout(context.Background(), c, t, time.Minute)
+	if err == nil {
+		t.Fatal("did not get expected error from server, got: nil")
+	}
+
+	want := fluent.ModifyError().
+		WithCode(codes.InvalidArgument).
+		AsStatus(t)
+
+	chk.HasRecvClientErrorWithStatus(t, err, want, chk.IgnoreDetails())
+}
+
+// InvalidElectionIDAndParams validates that the server returns an error when a client
+// specifies an update election ID at the same time as specifying session parameters (which
+// are illegal after the first message).
+func InvalidElectionIDAndParams(c *fluent.GRIBIClient, t testing.TB, _ ...TestOpt) {
+	defer electionID.Inc()
+	c.Connection().WithRedundancyMode(fluent.ElectedPrimaryClient).WithPersistence().WithInitialElectionID(electionID.Load(), 0)
+	c.Start(context.Background(), t)
+	defer c.Stop(t)
+
+	c.StartSending(context.Background(), t)
+
+	// Inject a specfic invalid entry that specifies election ID in two places along
+	// with the session parameters.
+	c.Modify().InjectRequest(t, &spb.ModifyRequest{
+		ElectionId: &spb.Uint128{
+			Low:  electionID.Load(),
+			High: 0,
+		},
+		Params: &spb.SessionParameters{
+			Redundancy:  spb.SessionParameters_ALL_PRIMARY,
```

Comment:
You forgot to change it to `SINGLE_PRIMARY` :) (the most implemented mode)

Reply:
Ugh, sorry -- fixing now...